doc/contributing/node-api-tooling.md

```diff
@@ -0,0 +1,179 @@
+# Node.js's API Documentation Tooling
+
+Node.js API documentation are generated by an in-house tooling that resides within the `tools/doc` directory. This tooling is composed of a few different pieces:
+
+1. The entry-point being `tools/doc/generate.js`.
+1. The tooling supports a few CLI arguments that is listed in a table below.
+1. The tooling processes one file at a time.
+1. The tooling uses a set of dependencies as described in the dependencies section.
+1. The tooling parses the input file and does several transformations to the AST (Abstract Syntax Tree).
+1. The tooling generates a JSON output that contains the metadata and content of the Markdown file
+1. The tooling generates a HTML output that contains a human-readable and ready to-view version of the file.
+
+These are the summarised steps of the API tooling. On the following sections we're diving deep what each part does, means and how it works.
+
+This documentation mainly serves the purpose of explaining the existing tooling processes, to allow an easier maintenance and evolution of the tooling. It is not meant to be a guide on how to write documentation for Node.js.
+
+#### Vocabulary & Good to Know's
+
+- AST means "Abstract Syntax Tree" and it is a data structure that represents the structure of a certain data format. In our case, the AST is a "graph" representation of the contents of the Markdown file.
+- MDN means "Mozilla Developer Network" and it is a website that contains documentation for web technologies. We use it as a reference for the structure of the documentation.
+- "Stability Index" is an internal concept for the Stability of a given Node.js module. There's no concrete documentation of the "Stability Index" but it goes as:
```

Comment:
```suggestion
 The [Stability Index](https://nodejs.org/dist/latest/docs/api/documentation.html#stability-index) is
 used to community the Stability of a given Node.js module.  The Stability levels include:
```

Reply:
Thanks! Didn't find a reference to them besides on the website.